controller/src/main/java/org/jboss/as/controller/ModelControllerClientFactoryImpl.java

```diff
@@ -122,7 +126,9 @@ private LocalClient(ModelController modelController, Supplier<SecurityIdentity>
 
         @Override
         public void close()  {
-            // whatever
+            threads.forEach(threadRef -> {Thread thread = threadRef.get(); if(thread != null) {
```

Comment:
Please use a for loop and save the class metadata the lambda adds.

Also, to be thread safe you have to wrap the iteration in a synchronized(threads) block.